R/calendly.R

```diff
@@ -1,37 +1,137 @@
 #' @return Calendly REST API response as a list
 #' @importFrom utils menu installed.packages
 #' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
 #' @export
 #' @examples \dontrun{
 #'
 #' authorize("calendly")
-#' get_calendly_user()
+#' token <- get_token(app_name = "calendly")
+#'
+#' result_list <- calendly_get(
+#'   url = "https://api.calendly.com/users/me",
+#'   token = token
+#' )
 #' }
-get_calendly_user <- function(api_key) {
-  # Get auth token
-  token <- get_token(app_name = "calendly")
-
-  # Declare URL
-  url <- "https://api.calendly.com/users/me"
-
-  # Github api get
+#'
+# Get Calendly stuffs
+calendly_get <- function(url, token, user = NULL, count = NULL, page_token = NULL) {
   result <- httr::GET(
     url,
+    query = list(
+      user = user,
+      count = count,
+      page = page_token
+    ),
     httr::add_headers(Authorization = paste0("Bearer ", token)),
     httr::accept_json()
   )
 
+  # Stop if error
   if (httr::status_code(result) != 200) {
     httr::stop_for_status(result)
   }
 
-  # Process and return results
   result_content <- httr::content(result, "text")
-  result_list <- jsonlite::fromJSON(result_content)
+  return(jsonlite::fromJSON(result_content))
+}
+
+#' Get Calendly API user
+#' @description This is a function to get the Calendly API user info
+#' @param api_key You can provide the API key directly or this function will attempt to grab an API key that was stored using the `authorize("calendly")` function
+#' @return Calendly REST API response as a list
+#' @importFrom utils menu installed.packages
+#' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
+#' @export
+#' @examples \dontrun{
+#'
+#' authorize("calendly")
+#' get_calendly_user()
+#' }
+get_calendly_user <- function(api_key) {
+
+  if (is.null(api_key)) {
+    # Get auth token
+    token <- get_token(app_name = "calendly")
+  } else {
+    token <- api_key
+  }
+  # Declare URL
+  result_list <- calendly_get(
+    url = "https://api.calendly.com/users/me",
+    token = token
+  )
+
   return(result_list)
 }
+
+#' Get Calendly Event Lists
+#' @description This is a function to get a list of scheduled events from a calendly user.
+#' @param api_key You can provide the API key directly or this function will attempt to grab an API key that was stored using the `authorize("calendly")` function
+#' @param user You need to retrieve the calendly user's URI. You can do this by doing `user <- get_calendly_user()` and `user$resource$uri`
+#' @param count The number of responses that should be returned. Default is 20 or you can say "all" to retrieve all.
+#' @return Calendly REST API response as a list
+#' @importFrom utils menu installed.packages
+#' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
+#' @export
+#' @examples \dontrun{
+#'
+#' authorize("calendly")
+#' user <- get_calendly_user()
+#' list_calendly_events(user = user$resource$uri)
+#'
+#' }
+#'
+list_calendly_events <- function(user, count = 100) {
```

Comment:
I think you forgot to add `api_key` as an argument:

```suggestion
list_calendly_events <- function(api_key, user, count = 100) {
```

Reply:
Added this missing argument.